recaf-ui/src/main/java/me/coley/recaf/ui/dialog/SsvmOptimizeDialog.java

```diff
@@ -106,7 +106,7 @@ public SsvmOptimizeDialog(CommonClassInfo owner, MethodInfo info, SsvmIntegratio
 				});
 				byte[] modified = writer.toByteArray();
 				// Replace in workspace
-				WorkspaceAPI.getPrimaryResource().getClasses().put(owner.getName(), ClassInfo.read(modified));
+				WorkspaceAPI.getPrimaryResource().getClasses().put(owner.getName(), ClassInfo.read(modified, owner.getSourceType()));
```

Comment:
Optimize will only ever operate on primary classes. Can inline and use `PRIMARY`.